core/ipc/rpc/kafka/src/main/java/org/opennms/core/ipc/rpc/kafka/KafkaRpcServerManager.java

```diff
@@ -287,8 +301,12 @@ public void run() {
                             }
                             // Should have complete message by this point.
                             final ByteString requestMessage = rpcContent;
+                            // Incoming RPC requests are gated to restrict number of threads used by Kafka RPC.
+                            // Need to process the requests after timeout even though permission acquisition fails.
```

Comment:
Can you elaborate on why we still need to process the request?

If we continue processing the request there may still be >maxConcurrentCalls active.

Reply:
I'm thinking this as gradual increase of threads instead of a burst that could cause issues with too many threads.  As per current defaults,  only 10 extra threads will be allowed per sec after a threshold of `maxConcurrentCalls`.   Currently burst only happens with `Provisiond` and this allows it to not to have a burst of requests within few secs.

Reply:
Makes sense - this will slow down and put back-pressure on the topic for bursts.

Let's also make sure to expose a Gauge for the actual number of threads active - since this may differ from the number of active calls being tracked by the bulkhead.